anyway/parsers/location_extraction.py

```diff
@@ -534,6 +544,16 @@ def extract_geo_features(db, newsflash: NewsFlash, update_cbs_location_only: boo
         for resolution in all_resolutions:
             if resolution not in location_from_db:
                 setattr(newsflash, resolution, None)
+    if newsflash.road_segment_name is not None and newsflash.road1 is not None:
```

Comment:
@ziv17 perhaps add another condition that newsflash.road_segment_id is not None as well?